tests/annotations/tests.py

```diff
@@ -160,6 +160,40 @@ def test_aggregate_over_annotation(self):
         other_agg = Author.objects.aggregate(age_sum=Sum('age'))
         self.assertEqual(agg['otherage_sum'], other_agg['age_sum'])
 
+    @skipUnlessDBFeature('can_distinct_on_fields')
+    def test_distinct_on_with_annotation(self):
+        store = Store.objects.create(
+            name='test store',
+            original_opening=datetime.datetime.now(),
+            friday_night_closing=datetime.time(21, 00, 00),
+        )
+        names = [
+            'Theodore Roosevelt',
+            'Eleanor Roosevelt',
+            'Franklin Roosevelt',
+            'Ned Stark',
+            'Catelyn Stark',
+        ]
+        for name in names:
+            Employee.objects.create(
+                store=store,
+                first_name=name.split()[0],
+                last_name=name.split()[1],
+                age=30, salary=2000,
+            )
+
+        people = Employee.objects.annotate(
+            name_lower=Func(F('last_name'), function='lower'),
```

Comment:
We could just use the `django.db.models.functions.Lower` function here, directly, without the Func wrapper. They produce exactly the same output though, so I'm not too concerned with this. Tim?